openmmtools/integrators.py

```diff
@@ -671,6 +672,206 @@ def __init__(self, temperature=298 * simtk.unit.kelvin, collision_rate=91.0 / si
         self.addConstrainVelocities()
 
 
+class NoseHooverChainVelocityVerletIntegrator(ThermostatedIntegrator):
+
+    """Nosé-Hoover chain thermostat, using the reversible multi time step velocity Verlet algorithm
+       detailed in the papers below.
+
+    References
+    ----------
+
+    G. J. Martyna, M. E. Tuckerman, D. J. Tobias, and Michael L. Klein  "Explicit reversible
+    integrators for extended systems dynamics", Molecular Physics, 87 1117-1157 (1996)
+    http://dx.doi.org/10.1080/00268979600100761
+
+    G. J. Martyna, M. L. Klein, and M. Tuckerman "Nosé–Hoover chains: The canonical
+    ensemble via continuous dynamics",
+    Journal of Chemical Physics 97, 2635-2643 (1992)
+    http://dx.doi.org/10.1063/1.463940
+
+    Examples
+    --------
+
+    Create a velocity Verlet integrator with Nosé-Hoover chain thermostat.
+
+    >>> timestep = 1.0 * simtk.unit.femtoseconds
+    >>> temperature = 300 * simtk.unit.kelvin
+    >>> chain_length = 10
+    >>> collision_frequency = 50 / simtk.unit.picoseconds
+    >>> num_mts = 5
+    >>> num_yoshidasuzuki = 5
+
+    >>> integrator = NoseHooverChainVelocityVerletIntegrator(temperature, collision_frequency, timestep, chain_length, num_mts, num_yoshidasuzuki)
+
+    Notes
+    ------
+
+    The velocity Verlet integrator is taken verbatim from Peter Eastman's example in the CustomIntegrator header file documentation.
+
+    Useful tests of the NHC integrator can be performed by monitoring the instantaneous temperature during the simulation and confirming that conserved energy is constant to about 1 part in 10^5.  The instantanous temperature and particle kinetic and potential energies can already be extracted from a snapshot, for example see the OpenMM StateDataReporter implementation for more details.  This integrator also provides heat bath energies (kJ/mol) through the following mechanism:
+
+    >>> integrator = NoseHooverChainVelocityVerletIntegrator()
+    >>> heat_bath_kinetic_energy = integrator.getGlobalVariableByName('bathKE')
+    >>> heat_bath_potential_energy = integrator.getGlobalVariableByName('bathPE')
+
+    From here, the conserved energy is the sum of the potential and kinetic energies for both the system and the heat bath.
+
+    """
+
+
+    YSWeights = {
+        1 : [ 1.0000000000000000 ],
+        3 : [ 0.8289815435887510, -0.6579630871775020,  0.8289815435887510 ],
+        5 : [ 0.2967324292201065,  0.2967324292201065, -0.1869297168804260, 0.2967324292201065, 0.2967324292201065 ]
+    }
+
+    def __init__(self, temperature=298*simtk.unit.kelvin, collision_frequency=50/simtk.unit.picoseconds,
```

Comment:
Can you change all the `simtk.unit` to just `unit`? I noticed at some point there was a double import at the top I can fix in a separate PR